tests/infra-scripts/overlapping_routes.py

```diff
@@ -0,0 +1,115 @@
+# s.* = switch id
+# > = signal
+# >> = signal that start a route ("Nf")
+#
+# This infra is used to test blocks that have overlapping routes.
+# There are 4 routes : a1->b1, a1->b2, a2->b1, a2->b2.
+# They all overlap on t.center
+
+
+# GENERATE INFRA
+builder = InfraBuilder()
+
+# Create operational points
+op_a1 = builder.add_operational_point("op.a1")
+op_a2 = builder.add_operational_point("op.a2")
+op_b1 = builder.add_operational_point("op.b1")
+op_b2 = builder.add_operational_point("op.b2")
+
+# Create track sections
+
+t_a1 = builder.add_track_section(length=1_000, label="t_a1")
+t_b1 = builder.add_track_section(length=1_000, label="t_b1")
+t_a2 = builder.add_track_section(length=1_000, label="t_a2")
+t_b2 = builder.add_track_section(length=1_000, label="t_b2")
+t_center = builder.add_track_section(length=10_000, label="t_center")
+
+# Add objects on tracks
+
+op_a1.add_part(t_a1, 0)
+op_b1.add_part(t_b1, 0)
+op_a2.add_part(t_a2, 1_000)
+op_b2.add_part(t_b2, 1_000)
+t_a1.add_buffer_stop(label="bf.a1", position=0)
+t_a2.add_buffer_stop(label="bf.a2", position=0)
+t_b1.add_buffer_stop(label="bf.b1", position=1_000)
+t_b2.add_buffer_stop(label="bf.b2", position=1_000)
+
+# Signals
+
+
+@dataclass
+class Signal:
+    name: str
+    track: Any
+    position: int
+    is_route_delimiter: bool
+
+
+raw_signals: List[Signal] = [
+    Signal("s.a1.nf", t_a1, 900, True),
+    Signal("s.a2.nf", t_a2, 900, True),
+    Signal("s.center.1", t_center, 4_000, False),
+    Signal("s.center.2", t_center, 6_000, False),
+    Signal("s.center.3", t_center, 8_000, False),
+    Signal("s.b1.nf", t_b1, 100, True),
+    Signal("s.b2.nf", t_b2, 100, True),
+]
+signals = []
+for raw_signal in raw_signals:
+    detector = raw_signal.track.add_detector(label=f"det.{raw_signal.name[2:]}", position=raw_signal.position)
+    signal = raw_signal.track.add_signal(
+        label=raw_signal.name,
+        position=raw_signal.position,
+        direction=Direction.START_TO_STOP,
+        is_route_delimiter=raw_signal.is_route_delimiter,
+    )
+    signal.add_logical_signal("BAL", settings={"Nf": "true" if raw_signal.is_route_delimiter else "false"})
+    signals.append(signal)
+
+# Add links
+
+s_a = builder.add_point_switch(t_center.begin(), t_a1.end(), t_a2.end(), label="s.a")
+s_b = builder.add_point_switch(t_center.end(), t_b1.begin(), t_b2.begin(), label="s.b")
+
+# Set coordinates
+
+lat_track_1 = 50
+lat_track_2 = 49.98
+
+t_a1.begin().set_coords(-0.12, lat_track_1)
+t_a1.end().set_coords(-0.1, lat_track_2)
+t_a2.begin().set_coords(-0.12, lat_track_2)
+t_a2.end().set_coords(-0.1, lat_track_2)
+
+t_center.begin().set_coords(-0.1, lat_track_2)
+t_center.end().set_coords(0.1, lat_track_2)
+
+t_b1.begin().set_coords(0.1, lat_track_2)
+t_b1.end().set_coords(0.12, lat_track_1)
+t_b2.begin().set_coords(0.1, lat_track_2)
+t_b2.end().set_coords(0.12, lat_track_2)
+
+# Build infra
+infra = builder.build()
+
+# Save railjson
+infra.save(OUTPUT_DIR / "infra.json")
```

Comment:
Do we have a process to make sure the json is not out of sync?

Reply:
We do, the CI runs the scripts and compare them with what's in the repo